lib/rdoc/task.rb

```diff
@@ -240,13 +240,13 @@ def define
     ].flatten.compact
 
     task rdoc_task_name => [rdoc_target]
+    args = option_list + @rdoc_files
     file rdoc_target => rdoc_target_deps do
       @before_running_rdoc.call if @before_running_rdoc
-      args = option_list + @rdoc_files
 
       $stderr.puts "rdoc #{args.join ' '}" if Rake.application.options.trace
-      RDoc::RDoc.new.document args
```

Comment:
This line could not be performed when it is inside the block passed as parameter to the "file" method. If really there is necessity of staying inside the "file" method, then there is something wrong in this method.